backend/src/test/java/codezap/like/service/LikesServiceTest.java

```diff
@@ -0,0 +1,127 @@
+package codezap.like.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import codezap.fixture.CategoryFixture;
+import codezap.fixture.MemberFixture;
+import codezap.fixture.TemplateFixture;
+import codezap.global.DatabaseIsolation;
+import codezap.global.ServiceTest;
+import codezap.member.domain.Member;
+import codezap.member.dto.MemberDto;
+import codezap.template.domain.Template;
+
+@SpringBootTest
+@DatabaseIsolation
+class LikesServiceTest extends ServiceTest {
+
+    @Autowired
+    private LikesService likesService;
+
+    @Nested
+    @DisplayName("좋아요 기능 테스트")
+    class LikesTest {
+
+        @Test
+        @DisplayName("성공")
+        void success() {
+            Member member = memberRepository.save(MemberFixture.getFirstMember());
+            Template template = templateRepository.save(TemplateFixture.get(
+                    member,
+                    categoryRepository.save(CategoryFixture.getFirstCategory())
+            ));
+
+            likesService.like(MemberDto.from(member), template.getId());
+
+            assertThat(likesRepository.existsByTemplateAndMember(template, member)).isTrue();
+        }
+
+        @Test
+        @DisplayName("성공: 동일한 사람이 동일한 템플릿에 여러번 좋아요를 해도 Likes 가 한번만 생성된다.")
+        void multipleLikes() {
+            Member member = memberRepository.save(MemberFixture.getFirstMember());
+            Template template = templateRepository.save(TemplateFixture.get(
+                    member,
+                    categoryRepository.save(CategoryFixture.getFirstCategory())
+            ));
+
+            likesService.like(MemberDto.from(member), template.getId());
+            likesService.like(MemberDto.from(member), template.getId());
+
+            assertThat(likesRepository.countByTemplate(template)).isEqualTo(1L);
+        }
+    }
+
+    @Nested
+    @DisplayName("좋아요 취소 기능 테스트")
+    class CancelLikesTest {
+
+        @Test
+        @DisplayName("성공")
+        void success() {
+            Member member = memberRepository.save(MemberFixture.getFirstMember());
+            Template template = templateRepository.save(TemplateFixture.get(
+                    member,
+                    categoryRepository.save(CategoryFixture.getFirstCategory())
+            ));
+            likesService.like(MemberDto.from(member), template.getId());
```

Comment:
이 부분도 `cancelLike`에 대한 테스트인데 `like`가 사용되는 것이 어색하게 느껴지네요. 아래와 같은 변경은 어떤가요??
```suggestion
            likesRepository.save(new Likes(null, template, member);
```

Reply:
그렇기도 하고, `Service` 계층 테스트 셋업을 하는데 자신의 다른 메서드를 사용할 필요는 없겠군요!
좋습니닷